### PR TITLE
Fix redeclarations if simpledb.h was included many times

### PR DESCRIPTION
**Pull request by q2012**
1. How to reproduce the error
Just create an "test.h" file in which include "simpledb.h" and in "db.c" include "test.h" and "simpledb.h".
The possible use case is when there are two "test.h", that implement some new features to "simpledb.h" and you want to use them, include "test1.h" and "test2.h" and get an error

2. Output of splint [splint_out.txt](https://github.com/thodnev/simpledb/files/2188073/splint_out.txt)

### CODE REVIEW

File: simpledb.c [hunk 1]
```diff
@@ -18,7 +18,7 @@ enum err item_read_f(FILE *fp, struct dbitem *item)
 	uint8_t buffer[sizeof(*item) + HASH_SIZE];
 	if (sizeof(buffer) != fread(buffer, 1, sizeof(buffer), fp))
 		return E_FREAD;
-	uLong crc_file = *(uLong *)buffer;
+	uLong crc_file = buffer[0] + (buffer[1] << 8) + (buffer[2] << 16) + (buffer[3] << 24); // was a warning
```
Comment (q2012) [inline simpledb.c]: This is awful but i dont know how to change it without opening new pull request
This is right variant:

uLong crc_file = 0;
for(size_t i = 0; i < HASH_SIZE; ++i) {
    crc_file += buffer[i];
}

But it is little-endian dependent

Comment (thodnev) [reply simpledb.c]: It's very good you care about endianness. But in fact the code `uLong crc_file = *(uLong *)buffer;` is endian-independent. Why? Because of the next process happening:

1. CRC is calculated with a help of zlib. Imagine you've calculated crc of file and it equals `0x12345678` on one platform and the `0x78563412` on the other (and also it should be transferred via Internet) -- a complete nonsense.
    So, zlib automatically takes care to be endian-independent.
2. It places its result in a `uLong` type which is generally a synonym to `long`
3. Long stores bytes either in big or little endian itself
4. Long is placed and retrieved exactly as-is, preserving the order, i.e.
    order
5. So what we here rely on? That bytes inside buffer and long **both** have the same order, that is guaranteed
And yes, files of our db won't be portable among big and little endian, but the code will. 

But I agree, this line deserves a good comment to avoid misunderstanding, i.e.
`uLong crc_file = *(uLong *)buffer;  /* We rely on zlib for endianness-independency */`

File: .gitignore [hunk 2]
```diff
@@ -39,3 +39,6 @@
 *.su
 *.idb
 *.pdb
+
+#local development
```
Comment (thodnev) [inline .gitignore]: This shouldn't be placed in public `.gitignore` as long as there's no profit for the community.
If for example you write tests and they rely on `local` directory used as `./db -d local`, then of course, it will be useful

**Comment by q2012** [comment]
First of all, my changes in all files, except gitignore and simpledb.h, were done because i had gcc version 7, which produced compile errors and warnings - which i fixed in PR. But after i updated my gcc to 8.1 - whole  project compiles fine. So there is no need in this changes, unless you use old compilers or want to be compatible with c99.

About include guards. I did a little research ([1](https://stackoverflow.com/questions/1804486/should-i-use-include-in-headers),[2](https://stackoverflow.com/questions/27810115/c-include-guards),[3](https://stackoverflow.com/questions/45526830/when-to-use-include-guards), [wiki](https://en.wikipedia.org/wiki/Include_guard)), and here`s what i found: 
1) Include guards have some problems, which couldn`t be solved by some "golden code standard".
2) Many compilers (not only c++) have [#pragma_once](https://en.wikipedia.org/wiki/Pragma_once), which meant to solve problem, but is not standard and has some issues ([conversation](https://stackoverflow.com/questions/1143936/pragma-once-vs-include-guards?noredirect=1&lq=1)).
3) Modern compilers recognize redefinitions to the same definition, and just ignore them. E. g. i tried 8.1 and could not reproduce the error.

About gitignore:  it is controverse, in my opinion, should there be a directory for local development and testing purposes, but what matters - it is definitely not obligatory.

To sum up: 
1) most changes were done just to build the project: switching to new compiler solves this. 
2) The error, this PR was meant to fix (redefinitions if header was included many times) could also be solved using modern compilers - as such, there is no need to solve this other way (with possible difficulties).

Having all this mentioned, i close PR.
